test/new-e2e/pkg/environments/aws/eks/eks.go

```diff
@@ -0,0 +1,414 @@
+
+// WorkloadAppFunc is a function that deploys a workload app to a kube provider
+type WorkloadAppFunc func(e config.CommonEnvironment, kubeProvider *kubernetes.Provider) (*kubeComp.Workload, error)
+
+// WithWorkloadApp adds a workload app to the environment
+func WithWorkloadApp(appFunc WorkloadAppFunc) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.workloadAppFuncs = append(params.workloadAppFuncs, appFunc)
+		return nil
+	}
+}
+
+// Run deploys a EKS environment given a pulumi.Context
+func Run(ctx *pulumi.Context, env *environments.EKS, params *ProvisionerParams) error {
+	var awsEnv aws.Environment
+	var err error
+	if env.AwsEnvironment != nil {
+		awsEnv = *env.AwsEnvironment
+	} else {
+		awsEnv, err = aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	clusterComp, err := components.NewComponent(*awsEnv.CommonEnvironment, awsEnv.Namer.ResourceName("eks"), func(comp *kubeComp.Cluster) error {
+		// Create Cluster SG
+		clusterSG, err := ec2.NewSecurityGroup(ctx, awsEnv.Namer.ResourceName("eks-sg"), &ec2.SecurityGroupArgs{
+			NamePrefix:  awsEnv.CommonNamer.DisplayName(255, pulumi.String("eks-sg")),
+			Description: pulumi.StringPtr("EKS Cluster sg for stack: " + ctx.Stack()),
+			Ingress: ec2.SecurityGroupIngressArray{
+				ec2.SecurityGroupIngressArgs{
+					SecurityGroups: pulumi.ToStringArray(awsEnv.EKSAllowedInboundSecurityGroups()),
+					PrefixListIds:  pulumi.ToStringArray(awsEnv.EKSAllowedInboundPrefixLists()),
+					ToPort:         pulumi.Int(22),
+					FromPort:       pulumi.Int(22),
+					Protocol:       pulumi.String("tcp"),
+				},
+				ec2.SecurityGroupIngressArgs{
+					SecurityGroups: pulumi.ToStringArray(awsEnv.EKSAllowedInboundSecurityGroups()),
+					PrefixListIds:  pulumi.ToStringArray(awsEnv.EKSAllowedInboundPrefixLists()),
+					ToPort:         pulumi.Int(443),
+					FromPort:       pulumi.Int(443),
+					Protocol:       pulumi.String("tcp"),
+				},
+			},
+			VpcId: pulumi.StringPtr(awsEnv.DefaultVPCID()),
+		}, awsEnv.WithProviders(config.ProviderAWS))
+		if err != nil {
+			return err
+		}
+
+		// Cluster role
+		clusterRole, err := localEks.GetClusterRole(awsEnv, "eks-cluster-role")
+		if err != nil {
+			return err
+		}
+
+		// IAM Node role
+		linuxNodeRole, err := localEks.GetNodeRole(awsEnv, "eks-linux-node-role")
+		if err != nil {
+			return err
+		}
+
+		windowsNodeRole, err := localEks.GetNodeRole(awsEnv, "eks-windows-node-role")
+		if err != nil {
+			return err
+		}
+
+		// Fargate Configuration
+		var fargateProfile pulumi.Input
+		if fargateNamespace := awsEnv.EKSFargateNamespace(); fargateNamespace != "" {
+			fargateProfile = pulumi.Any(
+				eks.FargateProfile{
+					Selectors: []awsEks.FargateProfileSelector{
+						{
+							Namespace: fargateNamespace,
+						},
+					},
+				},
+			)
+		}
+
+		// Create an EKS cluster with the default configuration.
+		cluster, err := eks.NewCluster(ctx, awsEnv.Namer.ResourceName("eks"), &eks.ClusterArgs{
```

Comment:
Actually I guess I should have asked, does this create a new cluster for each test run?

Reply:
at the moment it will create and destroy a cluster each time, we will not leak a cluster
it just the test will be slow to setup/cleanup (20 min each)
We discuss offline with @pducolin and we will probably use only one cluster for NPM and container team later, we plan to have a chat in Paris in person next week about that.